plugin/src/main/java/io/jenkins/plugins/analysis/core/steps/IssuesRecorder.java

```diff
@@ -703,7 +702,8 @@ void publishResult(final Run<?, ?> run, final TaskListener listener, final Strin
                 reportName, referenceJobName, referenceBuildId, ignoreQualityGate, ignoreFailedBuilds,
                 getSourceCodeCharset(),
                 new LogHandler(listener, loggerName, report.getReport()), statusHandler, failOnError);
-        publisher.attachAction(trendChartType);
+        ResultAction action = publisher.attachAction(trendChartType);
+        publisher.publishChecks(action);
```

Comment:
I looked into how safe this is,
It seems like it in the current state it will only work if all builds are using a github app.

If the API and implementation was separated out to different plugins (which I think it needs to be so that warnings-ng doesn't have to depend on github-branch-source) then it would fall back to a NullChecksPublisher and it would be far safer. In that case it would only fail for users with the github-checks plugin installed when they don't have a github app configured.

Thoughts?

Reply:
Not all builds, but builds from a github-branch-source project, for other project builds, they will still get `NullChecksPublisher` which is safe to use.

But for the stable use of warnings-plugin (since it is going to be used for ci.jenkin.io), maybe it's better to split the plugin before releasing this feature for warnings-ng? @uhafner

Reply:
We cannot release the changes in the warning plugin before we do an official release of the Checks API to the update center anyway. 

Shouldn't the `ChecksPublisherFactory` return a `NullPublisher` if the GitHub App is not configured for the run?

Reply:
ah, currently not, but it can be easily done if it is safer.

Reply:
yes it should be done